File: Server/src/HealingManager.ts
import { ProfileHelper } from "@spt/helpers/ProfileHelper"
import { AbstractModManager } from "./AbstractModManager"
import { Constants } from "./Constants"
import { IBodyPartHealth } from "@spt/models/eft/common/tables/IBotBase"

export class HealingManager extends AbstractModManager
{
    protected configName: string = "HealingConfig"

    public onPlayerDied(sessionId: string): void
    {
        if (this.config.enabled != true)
        {
            return
        }

        this.healPlayer(sessionId)

        Constants.getLogger().info(`${Constants.ModTitle}: Player healed!`)
    }

    private healPlayer(sessionId: string): void
    {
        const profile = Constants.Container.resolve<ProfileHelper>("ProfileHelper").getPmcProfile(sessionId)

        for (let partKey in profile.Health.BodyParts)
        {
            const part: IBodyPartHealth = profile.Health.BodyParts[partKey]

            part.Health.Current = part.Health.Maximum

            part.Effects = {}
        }
    }
}

File: Server/VAI-RLTP/Managers/HealingManager.cs
using SPTarkov.DI.Annotations;
using SPTarkov.Server.Core.Models.Common;
using SPTarkov.Server.Core.Models.Eft.Common;
using SPTarkov.Server.Core.Models.Eft.Common.Tables;

namespace VAI.RLTP.Managers;

[Injectable(InjectionType.Singleton)]
public sealed class HealingManager : AbstractModManager
{
    protected override string ConfigName => "HealingConfig";

    public void OnPlayerDied(MongoId sessionId)
    {
        if (!IsEnabled())
        {
            return;
        }

        HealPlayer(sessionId);
        Constants.GetLogger().Info($"{Constants.ModTitle}: Player healed!");
    }

    private void HealPlayer(MongoId sessionId)
    {
        var profile = ModContext.Current.ProfileHelper.GetPmcProfile(sessionId);
        if (profile is null)
        {
            return;
        }

        var health = profile.Health;
        if (health?.BodyParts is null)
        {
            return;
        }

        foreach (var part in health.BodyParts.Values)
        {
            if (part?.Health != null)
            {
                part.Health.Current = part.Health.Maximum;
            }

            part.Effects = new Dictionary<string, BodyPartEffectProperties>();
        }
    }
}
